atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java

```diff
@@ -721,11 +738,7 @@ private CompletableFuture<Channel> bootstrapClient(final Address address) {
     bootstrap.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 1000);
     bootstrap.group(clientGroup);
     bootstrap.channel(clientChannelClass);
-    bootstrap.resolver(
```

Comment:
❓ So we created on each client a new DNSResolver which opened then a socket got I this right? We now create one and reuse this.

Reply:
Plus it was not closed before.

Reply:
Exactly, the problem was it was never closed. The reusability is just because it makes sense to me, there doesn't seem to be any advantage in creating a new one every time.